backend/githubPlugin.js

```diff
@@ -0,0 +1,45 @@
+// @flow
+
+import type {ID, GraphNode, GraphEdge} from "./graph";
+
+export const GITHUB_PLUGIN_NANE = "sourcecred/github-beta";
+
+export type NodeType = "PULL_REQUEST" | "ISSUE" | "COMMENT" | "USER";
+
+export type PullRequestPayload = {
+  title: string,
+  isClosed: boolean,
+  isMerged: boolean,
+  dateCreated: number,
+  dateMerged: ?number,
+};
+
+export type IssuePayload = {
+  title: string,
+  isClosed: boolean,
+  dateCreated: number,
+};
+
+export type ReviewState = "APPROVED" | "COMMENTED" | "CHANGES_REQUESTED";
+export type CommentPayload = {
+  body: string,
+  date: number,
+  state: ?ReviewState,
+};
+
+export type UserPayload = {
+  githubUserId: number,
+};
+
+export type GithubNodePayload = {
+  type: NodeType,
+  subpayload: PullRequestPayload | IssuePayload | CommentPayload | UserPayload,
+};
+
+export type EdgeType = "AUTHOR" | "REFERENCE";
```

Comment:
As discussed: inclusion and co-inclusion will both fall under the `REFERENCE` type. That is, an issue REFERENCEs its comments, and the comments also REFERENCE their enclosing issues (resp. PR). We don’t see any reason why this should cause ambiguity, because an issue cannot otherwise reference a comment: it has no body in which to do so.

We’re punting on close/reopen actions for now.